README.md

```diff
@@ -1,3 +1,166 @@
+
+`decorator`: A react element positioned in the center of rendered `Node`. A decorator is similar to `component` in that it's a [render prop](https://reactjs.org/docs/render-props.html) that recieves the props of a the node, and returns a react element. This element however doesn't need to be an svg.
+
+```js
+const Decorator = props => {
+  return (
+    <button
+      style={decoratorStyles}
+      onClick={() => console.log(`You clicked ${props.label}`}
+    >
+      Click Me
+    </button>
+  );
+};
+
+// Use your render prop and custom props passed to `Node` to
+// cause differnet actions in your Decorator
+const MyNetwork = () => (
+  <Network>
+    <Node id='vadar' label='Darth Vadar' decorator={Decorator} />
+    <Node id='luke' label='Luke Skywalker' decorator={Decorator} />
+    <Node id='leia' label='Leia Organa' decorator={Decorator} />
+    <Edge id='1' from='vadar' to='luke' />
+    <Edge id='1' from='vadar' to='leia' />
+  </Network>
+);
+```
+
+**NB:** Because this is a positioned html element, it can recieve form events like `clicking` and inputs. While fairly performant in positioning and rendering, it can be straining if many nodes have decorators and should be used sparingly. For static (non-interactable) content, it would be better to use a `component` with a changing prop to re-render it.
+
+### < Edge />
+An `Edge` represents a [vis.js edge](http://visjs.org/docs/network/edges.html) and connects nodes together.
+
+Edges accept all of the [options](http://visjs.org/docs/network/edges.html#optionTable) vis exposes for edges as props. One specialy note, vis doesn't require that edges have an `id`, react-vis-network does in order to keep the network updated.
+
+### < ClusterByConnection />
+A `ClusterByConnection` is a representation of a [vis.js cluster](http://visjs.org/docs/network/#methodClustering). This particular cluster implementation collapses all nodes around, and including, it's `rootNodeId` into a cluster.
+
+`ClusterByConnection` takes all props that `Node` does including `component` and `decorator`.
+
+`rootNodeId`: `id` of a `Node` for which to cluster around.
+
+### Performance Considerations
+All exported components are [Pure Components](https://reactjs.org/docs/react-api.html#reactpurecomponent). It's important to not define anonymous functions and objects inside the render function as it will cause unnessisary re-renders.
+
+```js
+// bad, re-renders every time
+render () {
+  return <Node component={() => <SomeIcon />} />
+}
+
+// good, only re-renders when other props change
+renderSomeIcon = () => <SomeIcon />
+
+render () {
+  return <Node component={this.renderSomeIcon} />
+}
+```
+
+Use `decorators` on `Node`s sparingly, if you can do it in the `component` you should.
+
+### Extending, custom clusters, and custom modules
+> ⚠️ Advanced topics, here there be dragons.
```

Comment:
`here there` => `here`

Reply:
Ah that moment that you realize you've been saying an idiom wrong this whole time.